tests/functional/testOnboarding.js

```diff
@@ -369,4 +396,366 @@ describe('Onboarding', function() {
+    assert.equal(addDevicesSelectedEvents.length, 0);
+
+    //Ensure morePrivacySelected event is recorded from the devices screen
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.STEP_PROG_BAR_PRIVACY_BUTTON.visible());
+    morePrivacySelectedEvents = await vpn.gleanTestGetValue("interaction", "morePrivacySelected", "main");
+    assert.equal(morePrivacySelectedEvents.length, 1);
+    let morePrivacySelectedEventsExtras = morePrivacySelectedEvents[0].extra;
+    assert.equal(devicesScreenTelemetryId, morePrivacySelectedEventsExtras.screen);
+
+    await advanceToSlide(2)
+
+    //Ensure dataUseSelected event is recorded from the devices screen
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.STEP_PROG_BAR_DATA_BUTTON.visible());
+    dataUseSelectedEvents = await vpn.gleanTestGetValue("interaction", "dataUseSelected", "main");
+    assert.equal(dataUseSelectedEvents.length, 1);
+    dataUseSelectedEventsExtras = dataUseSelectedEvents[0].extra;
+    assert.equal(devicesScreenTelemetryId, dataUseSelectedEventsExtras.screen);
+
+
+    //Advance to start slide
+    await advanceToSlide(3)
+    await vpn.gleanTestReset();
+
+    //Start slide progress bar button does not have telemetry (because it never does anything)
+    //so we don't need to ensure it doesn't generate telemetry
+
+    //Ensure addDevicesSelected event is recorded from the start screen
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.STEP_PROG_BAR_DEVICES_BUTTON.visible());
+    addDevicesSelectedEvents = await vpn.gleanTestGetValue("interaction", "addDevicesSelected", "main");
+    assert.equal(addDevicesSelectedEvents.length, 1);
+    let addDevicesSelectedEventsExtras = addDevicesSelectedEvents[0].extra;
+    assert.equal(startScreenTelemetryId, addDevicesSelectedEventsExtras.screen);
+
+    await advanceToSlide(3)
+
+    //Ensure morePrivacySelected event is recorded from the start screen
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.STEP_PROG_BAR_PRIVACY_BUTTON.visible());
+    morePrivacySelectedEvents = await vpn.gleanTestGetValue("interaction", "morePrivacySelected", "main");
+    assert.equal(morePrivacySelectedEvents.length, 1);
+    morePrivacySelectedEventsExtras = morePrivacySelectedEvents[0].extra;
+    assert.equal(startScreenTelemetryId, morePrivacySelectedEventsExtras.screen);
+
+    await advanceToSlide(3)
+
+    //Ensure dataUseSelected event is recorded from the start screen
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.STEP_PROG_BAR_DATA_BUTTON.visible());
+    dataUseSelectedEvents = await vpn.gleanTestGetValue("interaction", "dataUseSelected", "main");
+    assert.equal(dataUseSelectedEvents.length, 1);
+    dataUseSelectedEventsExtras = dataUseSelectedEvents[0].extra;
+    assert.equal(startScreenTelemetryId, dataUseSelectedEventsExtras.screen);
+
+  });
+
+  it('Data slide events are recorded', async () => {
+    await vpn.gleanTestReset();
```

Comment:
It shouldn't be necessray to call this at the begining of test cases. This is called on the `beforeEach` block.

Reply:
Authentication is completed in the `beforeEach` of `testOnboarding`, which triggers a `continueSelected` event that we also want to test in `testOnboarding`

But you're right, I probably only need it at the beginning of the test that actually tests `continueSelected`

Reply:
Got it. If strictly necessary then fine, but do add a comment on why it was necessary to use this function.